### PR TITLE
Making pull activity timeout configurable in Docker

### PR DESCRIPTION
**Pull request by schledererj**

### CODE REVIEW

File: drivers/docker/config.go [hunk 1]
```diff
@@ -636,6 +645,14 @@ func (d *Driver) SetConfig(c *base.Config) error {
 		d.config.GC.DanglingContainers.CreationGrace = dur
 	}
 
+	if len(d.config.PullActivityTimeout) > 0 {
+		dur, err := time.ParseDuration(d.config.PullActivityTimeout)
+		if err != nil {
+			return fmt.Errorf("failed to parse 'pull_activity_timeout' duaration: %v", err)
+		}
```
Comment (notnoop) [inline drivers/docker/config.go]: Let's add a minimum activity timeout too, at least to rule out negative and zero values.  Would 1 minute be an appropriate floor?

File: drivers/docker/config.go [hunk 2]
```diff
@@ -553,16 +560,18 @@ type ContainerGCConfig struct {
...
+	TLS                         TLSConfig     `codec:"tls"`
+	GC                          GCConfig      `codec:"gc"`
+	Volumes                     VolumeConfig  `codec:"volumes"`
+	AllowPrivileged             bool          `codec:"allow_privileged"`
+	AllowCaps                   []string      `codec:"allow_caps"`
+	GPURuntimeName              string        `codec:"nvidia_runtime"`
+	InfraImage                  string        `codec:"infra_image"`
+	DisableLogCollection        bool          `codec:"disable_log_collection"`
+	PullActivityTimeout         string        `codec:"pull_activity_timeout"`
+	PullActivityTimeoutDuration time.Duration `codec:"-"`
```
Comment (notnoop) [inline drivers/docker/config.go]: The parsed duration can be a private field which would ensure that codec ignores it.

```suggestion
	pullActivityTimeoutDuration time.Duration `codec:"-"`
```

File: drivers/docker/config.go (hunk 1 above)
Comment (schledererj) [reply drivers/docker/config.go]: That sounds good, thanks. I'll add that. New commits incoming soon.

File: drivers/docker/config.go (hunk 2 above)
Comment (schledererj) [reply drivers/docker/config.go]: Awesome, that makes more sense. I'll make that change.

File: README.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ Resources
 -------------------------------
 
 * Website
-  * [www.nomadproject.io](https://www.nomadproject.io)
+  * [www.nomadproject.io](https://www.nomadproject.io)
```
Comment (jippi) [inline README.md]: 😲

File: drivers/docker/config.go [hunk 4]
```diff
@@ -258,6 +258,13 @@ var (
 			hclspec.NewLiteral(`"gcr.io/google_containers/pause-amd64:3.0"`),
 		),
 
+		// the duration that the driver will wait for activity from the Docker engine during an image pull
+		// before canceling the request
+		"pull_activity_timeout": hclspec.NewDefault(
+			hclspec.NewAttr("pull_activity_timeout", "string", false),
+			hclspec.NewLiteral(`"2m"`),
```
Comment (jippi) [inline drivers/docker/config.go]: is "activity" just any feedback on progress, or "docker pull need to complete within this time"?

Comment (schledererj) [reply drivers/docker/config.go]: Just any feedback on progress. I've noticed that with Windows images, sometimes the Docker engine will become busy for >2m when extracting a large-ish (~1GB) layers, and Nomad times out the pull.

File: README.md (hunk 3 above)
Comment (schledererj) [reply README.md]: Hah, that was just to re-trigger the CircleCI pipeline. I'll reverse it.

**Comment by schledererj** [comment]
Alrighty, I just signed the CLA, and it looks like the CI pipeline is passing. Let me know if you'd like me to make any more changes. Thanks!

**Pull request merged**